spaceship/lib/spaceship/portal/certificate.rb

```diff
@@ -292,17 +309,26 @@ def create!(csr: nil, bundle_id: nil)
           mac = MAC_CERTIFICATE_TYPE_IDS.include?(type)
 
           # look up the app_id by the bundle_id
-          if bundle_id
+          if bundle_id && !bundle_id.start_with?('merchant.')
```

Comment:
maybe better add an additional inner `if` instead of this strange combination of conditions? I think currently this is buggy if there actually is no `bundle_id`

Something like this maybe?
```
if bundle_id
   if !bundle_id.start_with?('merchant.')
      ...
   else
      ...
   end
end
```

Reply:
made the change.